src/bun.js/api/bun/socket.zig

```diff
@@ -1880,6 +1954,104 @@ fn NewSocket(comptime ssl: bool) type {
             return JSValue.jsUndefined();
         }
 
+        pub fn getTLSTicket(
+            this: *This,
+            globalObject: *JSC.JSGlobalObject,
+            _: *JSC.CallFrame,
+        ) callconv(.C) JSValue {
+            if (comptime ssl == false) {
+                return JSValue.jsUndefined();
+            }
+
+            if (this.detached) {
+                return JSValue.jsUndefined();
+            }
+
+            var ssl_ptr: *BoringSSL.SSL = @ptrCast(*BoringSSL.SSL, this.socket.getNativeHandle());
+            const session = BoringSSL.SSL_get_session(ssl_ptr) orelse return JSValue.jsUndefined();
+            var ticket: [*c]const u8 = undefined;
+            var length: usize = 0;
+            BoringSSL.SSL_SESSION_get0_ticket(session, @ptrCast([*c][*c]const u8, &ticket), &length);
```

Comment:
What is the lifetime of this value? Do we need to free it? Note that `.createBuffer` will clone the contents of the ticket, which means if this is allocated then `ticket` must be freed at the end of this scope.

Reply:
basically from the docs`The pointer is only valid while the connection is in use.` the cloning is intentional.

Reply:
can you add that as a comment there so we know that later?

Reply:
fix on https://github.com/oven-sh/bun/commit/927f54269bdf489d21be4a8a6b89b448fdf48184